### PR TITLE
tls ca fix

### LINKED ISSUE #59: self signed ssl bug
There is currently a bug in the self signed CA chain which is causing issues in TLS koji-hub connectivity.

## Acceptance Criteria

* "child certificates" should be validated against  the root server CA

## Definition of Done

A valid certificate chain between client and server.

### PR DESCRIPTION
**Pull request by odra**
fixes #59 

## Changes

* re-factors tls cert generation to include CA extensions
* adds psql schema import in koji-hub image
* fixes readiness koji-hub script

NOTE: you will need to build both koji-hub and operator images on your own and change the image reference before deployment or use the following built images:

* quay.io/lrossett/mbox-operator:latest
* quay.io/lrossett/koji-hub:latest

## Verification steps

* Deploy mbox operator and kojihub cr
* "ssh" into the koji-hub pod: `k exec -it koji-hub-6bff9b7544-4jwq7 -- /bin/bash`
* cd into `/tmp` and run the following commands:
  * `openssl genrsa -out client.key 4096`
  * `openssl req -new -key client.key -out client.csr` (skip everything but common name which should be the desired koji username such as "mboxdev")
  * `openssl x509 -req -in client.csr -CA /etc/cacert/cert -CAkey /etc/cacert/key -CAcreateserial -CAserial client.srl -days 730 -sha256 -out client.crt`
  * Run `openssl verify client.crt`, expected output: `client.crt OK`
  * create a `data.xml` file:
    ```xml
    <?xml version="1.0" encoding="utf-8"?>
    <methodCall>
      <methodName>sslLogin</methodName>
      <params>
      </params>
    </methodCall>
    ```
  * run the command: `curl -k -X POST --cert client.crt --key client.key  -d @data.xml https://127.0.0.1:8443/kojihub/ssllogin`, expected output:
    ```xml
    <?xml version='1.0'?>
    <methodResponse>
    <params>
    <param>
    <value><struct>
    <member>
    <name>session-id</name>
    <value><int>1</int></value>
    </member>
    <member>
    <name>session-key</name>
    <value><string>1-4avhfEn8INv0uvUGduA</string></value>
    </member>
    </struct></value>
    </param>
    </params>
    </methodResponse>
    ```

### CODE REVIEW

File: images/koji-hub/test.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+X=/home/lrossett
```
Comment (Zlopez) [inline images/koji-hub/test.sh:1]: This looks like some test script and you just forgot to remove it when creating PR.

File: mbox-operator/roles/koji-hub/tasks/cert.yml [hunk 2]
```diff
@@ -1,35 +1,49 @@
...
-        privatekey_path: /tmp/{{ cert_name }}_server_key.pem
+        path: "{{ cert_dir.path }}/server_req.pem"
+        privatekey_path: "{{ cert_dir.path }}/server_key.pem"
         common_name: "{{ koji_hub_host }}"
     - openssl_certificate:
-        path: /tmp/{{ cert_name }}_server_cert.crt
-        csr_path: /tmp/{{ cert_name }}_server_req.csr
-        ownca_path: /tmp/{{ cert_name }}_ca_cert.crt
-        ownca_privatekey_path: /tmp/{{ cert_name }}_ca_key.pem
+        path: "{{ cert_dir.path }}/server_cert.pem"
```
Comment (Zlopez) [inline mbox-operator/roles/koji-hub/tasks/cert.yml:43]: Why PEM certificate everywhere?

File: mbox-operator/roles/koji-hub/templates/configmap.yml.j2 [hunk 3]
```diff
@@ -127,14 +127,19 @@ data:
     NotifyOnSuccess = False
 
     Plugins = fedmsg-koji-plugin runroot_hub
+
+    MaxRequestLength = 2147483647
+    KojiDebug = True
```
Comment (Zlopez) [inline mbox-operator/roles/koji-hub/templates/configmap.yml.j2:132]: The debug should be removed from config or added only for testing.

File: mbox-operator/roles/koji-hub/templates/configmap.yml.j2 [hunk 4]
```diff
@@ -127,14 +127,19 @@ data:
...
+    KojiTraceback = extended
   kojiweb.conf: |-
     [web]
     SiteName = MBOX Koji
     KojiHubURL = https://{{ koji_hub_host }}:8443/kojihub
     KojiFilesURL = https://{{ koji_hub_host }}/pkgs
     Secret = MboxKojiWeb
     LibPath = /usr/share/koji-web/lib
-    LoginDisabled = True
+    LoginDisabled = False
```
Comment (Zlopez) [inline mbox-operator/roles/koji-hub/templates/configmap.yml.j2:142]: Why this change?

File: mbox-operator/roles/koji-hub/tasks/cert.yml (hunk 2 above)
Comment (odra) [reply mbox-operator/roles/koji-hub/tasks/cert.yml:43]: hey are all in pem format by default, I thought using "pem" would be more expressive.

**Comment by Zlopez** [comment]
One more question: In the description you wrote:

> NOTE: you will need to build both koji-hub and operator images on your own and change the image reference before deployment or use the following built images:
>
>    quay.io/lrossett/mbox-operator:latest
>    quay.io/lrossett/koji-hub:latest

Why we can't use fedora namespace?

**Comment by odra** [comment]
because I don't want to use fedora org for my dev images

**Comment by Zlopez** [comment]
Isn't koji-hub already there?

**Comment by odra** [comment]
yes, but I made changes that will not be commited to quay.io until we merge the pr.

**Comment by Zlopez** [comment]
Ok, so the note was just for testing this PR.

**Comment by odra** [comment]
yeah just a note for testing this pr :)
